### ISSUE TITLE
sometimes get no results

### ISSUE THREAD

**Issue by chaosmonk1** [searx/searx#1792]
Hi, I'm running Searx 0.15.0 as packaged by Debian.  About 50% of the time I attempt a search I get no results, even if the exact same query did return results immediately before or after.  I don't have Google enabled as a search engine, so that's not the issue.  I also don't think it is a timeout issue, because in `settings.yml` I have timeout set to 30 seconds globally and unset for all enabled search engines, and yet the failed searches take only a few seconds.

The debug log provides the same output whether or not a search succeeds or fails.  For example, here is the output after a search which returned results only from Bing:
```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): en.wikipedia.org:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.bing.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.qwant.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): startpage.com:443
DEBUG:urllib3.connectionpool:https://www.bing.com:443 "GET /search?q=language%3AEN+testing+search+engine&first=1 HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://en.wikipedia.org:443 "GET /w/api.php?action=query&format=json&titles=testing+search+engine%7CTesting+Search+Engine&prop=extracts%7Cpageimages&exintro&explaintext&pithumbsize=300&redirects HTTP/1.1" 200 146
DEBUG:urllib3.connectionpool:https://api.qwant.com:443 "GET /api/search/web?count=10&offset=0&f=&q=testing+search+engine&t=web&uiv=4&locale=en_us HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://startpage.com:443 "POST /do/search HTTP/1.1" 200 None
```

And here is the apparently identical output after a search which returned no results:
```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): en.wikipedia.org:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.bing.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.qwant.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): startpage.com:443
DEBUG:urllib3.connectionpool:https://www.bing.com:443 "GET /search?q=language%3AEN+testing+search+engine&first=1 HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://en.wikipedia.org:443 "GET /w/api.php?action=query&format=json&titles=testing+search+engine%7CTesting+Search+Engine&prop=extracts%7Cpageimages&exintro&explaintext&pithumbsize=300&redirects HTTP/1.1" 200 146
DEBUG:urllib3.connectionpool:https://api.qwant.com:443 "GET /api/search/web?count=10&offset=0&f=&q=testing+search+engine&t=web&uiv=4&locale=en_us HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://startpage.com:443 "POST /do/search HTTP/1.1" 200 None
```

I'm not sure how next to troubleshoot.  Can anyone provide some guidance?  Thanks.

**Comment by return42** [comment]
>  Searx 0.15.0 as packaged by Debian

Sorry, can't help with this installation method (it missed a lot of bugfixes), please use our [installation instruction](https://asciimoo.github.io/searx/admin/installation.html)

**Comment by chaosmonk1** [comment]
> >  Searx 0.15.0 as packaged by Debian
> 
> Sorry, can't help with this installation method (it missed a lot of
> bugfixes), please use our [installation
> instruction](https://asciimoo.github.io/searx/admin/installation.html)

Thanks for your reply.  Is it the "Searx 0.15.0" part or the "as
packaged by Debian" part that you have a problem with?  0.15.0 is the
most recent version of Searx.[1]  If I must use git instead of the most
current release then I'm willing to do so, but in that case I don't
understand why you have releases at all.  Or is there a newer release
than 0.15.0 available somewhere other than Github?

[1] https://github.com/asciimoo/searx/releases

**Comment by return42** [comment]
Both :)

From time to time we set a release tag, but we do not maintain patched releases. For want on resources this is unfortunately not possible.  Thats why we always recommend to follow installation instruction and the https://asciimoo.github.io/searx/admin/installation.html#how-to-update guide.

I guess we will never have patched releases. Because searx  addresses a moving target (parsing changed result lists from origin search engines) "breaking bugs" could pop up every day and has to be fixed ASAP.  Mostly those fixes are contributed from searx admins.

Searx is nothing you can install and use for a long time without patching.  You have to maintain your searx instance and monitor the issue tracker.  BTW if your instance is public, you have to [protect](https://asciimoo.github.io/searx/admin/filtron.html) searx. If you are more an end-user than a searx admin, we recommend to use one of the publich searx instance ...

"""Get started with searx by using one of the Searx-instances. If you don’t trust anyone, you can set up your own, see Installation."""" [1]

[1] https://asciimoo.github.io/searx/

**Comment by chaosmonk1** [comment]
I see, thanks for the information.  I removed the Debian package and installed Searx from git.  So far it is performing much better.  Thanks.  I did run into a couple of issues with the official instructions:
* In Debian/Ubuntu, `virtualenv` is provided by the "virtualenv" package, not "python-virtualenv".  When I ran `virtualenv searx-ve` I got `virtualenv: not found` until I installed the virtualenv package.
* Whereas Debian's instructions for installing Searx use Python 3, the official instructions use Python 2.  Python 2 is EOL and no longer receives security updates.
